### ISSUE TITLE
Jupyter codemirror autoLoadMode bug?

### ISSUE THREAD

**Issue by BoPeng** [jupyter/notebook#3467]
I am trying to customize codemirror mode for some cells in my Jupyter extension. Setting

```
cell.user_highlight
```
triggers some errors saying 
```
Exception in event handler for edit_mode.Cell 
TypeError: CodeMirror.autoLoadMode is not a function
    at CodeCell.Cell._auto_highlight (main.min.js?v=11b42e90f007d9034055418999cda73c:23231)
    at CodeCell.Cell.auto_highlight (main.min.js?v=11b42e90f007d9034055418999cda73c:23213)
    at CodeCell.select (main.min.js?v=11b42e90f007d9034055418999cda73c:31175)
    at Notebook.select (main.min.js?v=11b42e90f007d9034055418999cda73c:37995)
    at Notebook._contract_selection (main.min.js?v=11b42e90f007d9034055418999cda73c:37967)
    at Notebook.handle_edit_mode (main.min.js?v=11b42e90f007d9034055418999cda73c:38091)
    at window._Events.<anonymous> (main.min.js?v=11b42e90f007d9034055418999cda73c:37478)
    at window._Events.dispatch (main.min.js?v=11b42e90f007d9034055418999cda73c:3)
    at window._Events.r.handle (main.min.js?v=11b42e90f007d9034055418999cda73c:3)
    at Object.trigger (main.min.js?v=11b42e90f007d9034055418999cda73c:4) Arguments(2)
```

The error happens at [this line of code](https://github.com/jupyter/notebook/blob/master/notebook/static/notebook/js/cell.js#L642). As far as I can tell, this function requires `codemirror/addon/mode/loadmode.js` which is not defined [here](https://github.com/jupyter/notebook/blob/master/notebook/static/notebook/js/cell.js#L19). Is this a bug?

**Comment by BoPeng** [comment]
To reproduce this bug:

1. Open any notebook
2. In browser console window, run
```
IPython.notebook.get_selected_cell().user_highlight = 'sas'
```
3. Click the cell

**Comment by wRosie** [comment]
I encountered the same problem here after two years. Have you found a solution to set codemirror mode without getting into this problem? Thanks!

**Comment by BoPeng** [comment]
I think I get around this bug but I have later shifted to a jupyterlab interface because classic jupyter notebook seemed to be unmaintenanced...

**Comment by kevin-bates** [comment]
You're right, this repository is undermaintained and we've [called that out in the README](https://github.com/jupyter/notebook#notice).  We could certainly use more help with maintaining the repo - particularly with respect to the front end.  

A good way to help would be to take a look at open pull requests that haven't been reviewed or are waiting on the author and provide reviews and/or continue working the pull request if it appears the author is inactive.

As for this issue, it looks like you were on the right track.  If there's a possible fix you feel would address the issue, please feel free to provide a pull request and we'll do our level best to get it reviewed.  Thank you.

**Comment by BoPeng** [comment]
@wRosie Could you work on a PR for this bug? I pointed out the problem in the original ticket which should be very easy to fix. I had to use a walkaround in my code but a PR would be a better way to proceed.

**Comment by wRosie** [comment]
I finally did a walkaround to avoid this problem, yet I am happy to work on a PR. It seems you are right on the problem and it should be an easy fix.
